bool is_start_generate
---
bool is_received
